tools/ci/tc/tests/test_valid.py

```diff
@@ -0,0 +1,72 @@
+import json
+import os
+
+import jsone
+import mock
+import pytest
+import requests
+import sys
+import yaml
+from jsonschema import validate
+
+here = os.path.dirname(__file__)
+root = os.path.abspath(os.path.join(here, "..", "..", "..", ".."))
+
+
+def data_path(filename):
+    return os.path.join(here, "..", "testdata", filename)
+
+
+@pytest.mark.xfail(sys.version_info.major == 2,
+                   reason="taskcluster library has an encoding bug")
+def test_verify_taskcluster_yml():
+    """Verify that the jsone in the .taskcluster.yml is valid"""
+    with open(os.path.join(root, ".taskcluster.yml")) as f:
+        template = yaml.safe_load(f)
+
+    events = [("pr_event.json", "github-pull-request", "Pull Request"),
+              ("master_push_event.json", "github-push", "Push to master")]
+
+    for filename, tasks_for, title in events:
+        with open(data_path(filename)) as f:
+            event = json.load(f)
+
+        context = {"tasks_for": tasks_for,
+                   "event": event,
+                   "as_slugid": lambda x: x}
+
+        jsone.render(template, context)
+
+
+def test_verify_payload():
+    """Verify that the decision task produces tasks with a valid payload"""
+    from tools.ci.tc.decision import decide
+
+    create_task_schema = requests.get(
+        "https://raw.githubusercontent.com/taskcluster/taskcluster/blob/master/services/queue/schemas/v1/create-task-request.yml")
```

Comment:
Does this need to be master, or should this be pinned to some commit to not have the tests break?